hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/processor/ProcessorSupplierWithService.java

```diff
@@ -50,7 +52,12 @@ private ProcessorSupplierWithService(
 
     @Override
     public void init(@Nonnull ProcessorSupplier.Context context) {
-        this.serviceContext = serviceFactory.createContextFn().apply(context);
+        HazelcastInstanceImpl hazelcastInstance = (HazelcastInstanceImpl) context.jetInstance().getHazelcastInstance();
```

Comment:
I think it's better to cast to `SerializationServiceSupport`. this will work with both `HazelcastInstanceImpl` and `HazelcastInstanceProxy`

Reply:
alternatively the processor could implement the `SerializationServiceAware` interface. I played with that here: https://github.com/hazelcast/hazelcast-jet/pull/1834/files

it does not require casting at all, but I am not sure whether it's really any better.

Reply:
For the processor it is too late. we create a single `serviceContext` per member here and pass it to the processors so that they create a service using this context. So we want the `managedContext` to initialize the `serviceContext` and do it once, after that we can create services using the injected values in the `serviceContext`